### PR TITLE
Makefile: add a DBG variable

### PR DESCRIPTION
**Pull request by thockin**
Now `make DBG=1` will produce binaries with no optimizaions and no inlining, but with symbols and DWARF information.

This also sets the various compile flags to use "all=" syntax.  Otherwise things like trimpath only apply to the CLI-listed packages and nothing else (which seems egregiously wrong).

Worth noting: this builds tools (like codegens) with debugging too.  We could make it more sophisticated if that seems a problem (e.g.  a different flag or value)

To discuss: "DBG" vs "KUBE_DBG" vs `make debug` vs ...

/kind cleanup

```release-note
NONE
```

### CODE REVIEW

**Comment by liggitt** [comment]
the error looks like a legitimate issue with the build flags

**Comment by thockin** [comment]
Flags errors seem to be fixed.  I had mis-understood the go docs before.  Better now.

File: build/root/Makefile.generated_files [hunk 1]
```diff
@@ -205,7 +206,8 @@ $(PRERELEASE_LIFECYCLE_FILES): $(PRERELEASE_LIFECYCLE_GEN)
 # newer than the binary, and try to "rebuild" it over and over.  So we touch
 # it, and make is happy.
 $(PRERELEASE_LIFECYCLE_GEN): $(GODEPS_k8s.io/kubernetes/vendor/k8s.io/code-generator/cmd/prerelease-lifecycle-gen)
-	KUBE_BUILD_PLATFORMS="" hack/make-rules/build.sh ./vendor/k8s.io/code-generator/cmd/prerelease-lifecycle-gen
+	KUBE_BUILD_PLATFORMS="" \
+	    hack/make-rules/build.sh vendor/k8s.io/code-generator/cmd/prerelease-lifecycle-gen
```
Comment (liggitt) [inline build/root/Makefile.generated_files:210]: huh, leading `./` got dropped?

Comment (liggitt) [reply build/root/Makefile.generated_files:210]: oh, this isn't go install, it was already calling build.sh

Comment (thockin) [reply build/root/Makefile.generated_files:210]: yeah, the ./ just shows up in the output in a weird way, and isn't strictly needed.  And actually, later in my go-worspaces tree, I think it changes again.  I could revert this part if you care.

Comment (thockin) [reply build/root/Makefile.generated_files:210]: oh, that part already merged in a different PR.  Still, I can undo it if it bothers :)

**Comment by thockin** [comment]
rebased

**Comment by thockin** [comment]
I broke the "all=" change to a distinct commit

**Pull request merged**